### PR TITLE
fix: quoting support

### PR DESCRIPTION
**Pull request by Jrmyy**
### Description

Closes #65 

## Models used to test - Optional
**`_iceberg_data_source.sql`**
```sql
{{ config(
  materialized='table',
  partitioned_by=['hour(ts)'],
  table_type='iceberg',
  table_properties={
    'vacuum_max_snapshot_age_seconds': '86400',
    'optimize_rewrite_delete_file_threshold': '2'
  }
) }}

SELECT cast(t.ts AS timestamp(6)) AS ts
FROM
    unnest(
        sequence(
            current_date - INTERVAL '1' DAY,
            cast(REPLACE(cast(current_timestamp as varchar), ' UTC', '') as timestamp(6)),
            INTERVAL '1' MINUTE
        )
    ) AS t(ts)
```

**`_iceberg_strat_append.sql`**
```sql
{{ config(
  materialized='incremental',
  incremental_strategy='append',
  partitioned_by=['hour(ts)'],
  table_type='iceberg',
  format='avro',
) }}

SELECT
    ts
    , 'abc' AS str
FROM {{ ref('_iceberg_data_source') }}
{% if is_incremental() %}
  -- this filter will only be applied on an incremental run
  WHERE ts > (SELECT max(ts) FROM {{ this }})
{% endif %}
```

**`_iceberg_strat_merge.sql`**
```sql
{{ config(
  materialized='incremental',
  incremental_strategy='merge',
  partitioned_by=['hour(ts)'],
  unique_key='ts',
  table_type='iceberg',
  format='orc',
) }}

SELECT
    ts
    , 'abc' AS str
FROM {{ ref('_iceberg_data_source') }}
{% if is_incremental() %}
  -- this filter will only be applied on an incremental run
  WHERE ts > (SELECT max(ts) FROM {{ this }})
{% endif %}
```

**`_parquet_data_source.sql`**
```sql
{{ config(
  materialized='table',
  format='json'
) }}

SELECT cast(t.ts AS timestamp(3)) AS ts
FROM
    unnest(
        sequence(
            current_date - INTERVAL '1' DAY,
            cast(REPLACE(cast(current_timestamp as varchar), ' UTC', '') as timestamp(3)),
            INTERVAL '1' MINUTE
        )
    ) AS t(ts)
```

**`_parquet_strat_append`**
```sql
{{ config(
  materialized='incremental',
  incremental_strategy='append',
  format='avro'
) }}

SELECT
    ts
    , 'abc' AS str
FROM {{ ref('_parquet_data_source') }}
{% if is_incremental() %}
  -- this filter will only be applied on an incremental run
  WHERE ts > (SELECT max(ts) FROM {{ this }})
{% endif %}
```

**`_parquet_strat_io.sql`**
```sql
{{ config(
  materialized='incremental',
  incremental_strategy='insert_overwrite',
  partitioned_by=['date_hour'],
) }}

SELECT
    ts
    , 'abc' AS str
    , date_trunc('hour', ts) AS date_hour
FROM
    {{ ref('_parquet_data_source') }}
{% if is_incremental() %}
    -- this filter will only be applied on an incremental run
    WHERE ts > (SELECT max(date_hour)
        FROM {{ this }}




    )
{% endif %}
```

**`_other_db.yaml`**
```yaml
version: 2

sources:
  - name: other_db
    database: other_db

    tables:
      - name: "abc def"
```

**`_ijk.sql`**
```sql
{{ config(
  materialized='table',
  alias='ijk lmn',
  table_type='iceberg',
  format='avro',
) }}

SELECT ts FROM {{ source('other_db', 'abc def') }}
```

## Checklist
- [X] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [X] You added unit testing when necessary
- [X] You added functional testing when necessary

### CODE REVIEW

**Comment by BrechtDeVlieger** [comment]
Hi @Jrmyy Thanks a lot for looking into this! I checked the code and it looks good. I also installed the branch and tested myself. I tried building tables that start with an underscore and it works like a charm. All the other models also still work as expected 👌

**Review by jessedobbelaere — DISMISSED** [review]

**Review by nicor88 — APPROVED** [review]
Code looks good. I didn't manage to test the feature.

File: dbt/include/athena/macros/adapters/schema.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{% macro default__create_schema(relation) -%}
```
Comment (BrechtDeVlieger) [inline dbt/include/athena/macros/adapters/schema.sql:1]: Shouldn't this be `athena__` prefix instead?

File: dbt/include/athena/macros/adapters/schema.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  {%- call statement('create_schema') -%}
+    create schema if not exists {{ relation.without_identifier().render_hive() }}
+  {% endcall %}
+{% endmacro %}
+
+
+{% macro default__drop_schema(relation) -%}
```
Comment (nicor88) [inline dbt/include/athena/macros/adapters/schema.sql:8]: nit: same as here, https://github.com/dbt-athena/dbt-athena/pull/152/files#r1112948055 maybe we prefix with `athena__`

File: dbt/include/athena/macros/adapters/schema.sql (hunk 1 above)
Comment (Jrmyy) [reply dbt/include/athena/macros/adapters/schema.sql:1]: Yes indeed I did not check all the code in the previous commits :)

File: dbt/include/athena/macros/adapters/schema.sql (hunk 2 above)
Comment (Jrmyy) [reply dbt/include/athena/macros/adapters/schema.sql:8]: Yes indeed I did not check all the code in the previous commits :)

**Comment by BrechtDeVlieger** [comment]
Looks good to me know 👍

**Review by jessedobbelaere — APPROVED** [review]

**Pull request merged**